packages/compass-components/src/components/document-list/document-actions-group.tsx

```diff
@@ -0,0 +1,141 @@
+import React, { useEffect, useRef, useState } from 'react';
+import { css } from '@leafygreen-ui/emotion';
+import { spacing } from '@leafygreen-ui/tokens';
+import { Button, Icon } from '../leafygreen';
+
+const actionsGroupContainer = css({
+  position: 'absolute',
+  display: 'flex',
+  gap: spacing[2],
+  width: '100%',
+  top: spacing[3],
+  paddingLeft: spacing[3],
+  paddingRight: spacing[3],
+  pointerEvents: 'none',
+});
+
+const actionsGroupItem = css({
+  flex: 'none',
+  pointerEvents: 'all',
+});
+
+const actionsGroupItemSeparator = css({
+  flex: '1 0 auto',
+});
+
+function useElementParentHoverState<T extends HTMLElement>(
+  ref: React.RefObject<T>
+): boolean {
+  const [isHovered, setIsHovered] = useState(false);
+
+  useEffect(() => {
+    const node = ref.current?.parentElement;
+
+    const onMouseEnter = () => {
+      setIsHovered(true);
+    };
+
+    const onMouseLeave = () => {
+      setIsHovered(false);
+    };
+
+    node?.addEventListener('mouseenter', onMouseEnter);
+    node?.addEventListener('mouseleave', onMouseLeave);
+
+    return () => {
+      node?.removeEventListener('mouseenter', onMouseEnter);
+      node?.removeEventListener('mouseleave', onMouseLeave);
+    };
+  }, [ref]);
+
+  return isHovered;
+}
+
+const DocumentActionsGroup: React.FunctionComponent<
+  {
+    onEdit?: () => void;
+    onCopy?: () => void;
+    onClone?: () => void;
+    onRemove?: () => void;
+    showOnHover?: boolean;
+  } & (
+    | { onExpand?: never; expanded?: never }
+    | { onExpand: () => void; expanded: boolean }
+  )
+> = ({
+  onEdit,
+  onCopy,
+  onClone,
+  onRemove,
+  onExpand,
+  expanded,
+  showOnHover = true,
+}) => {
+  const conatinerRef = useRef<HTMLDivElement | null>(null);
+  const isHovered = useElementParentHoverState(conatinerRef);
+
+  return (
+    <div
+      ref={conatinerRef}
+      className={actionsGroupContainer}
+      style={{ display: showOnHover ? (isHovered ? 'flex' : 'none') : 'flex' }}
+    >
+      {onExpand && (
+        <Button
+          size="xsmall"
+          rightGlyph={
+            <Icon glyph={expanded ? 'CaretDown' : 'CaretRight'}></Icon>
+          }
+          title={expanded ? 'Collapse all' : 'Expand all'}
+          aria-pressed={expanded}
+          data-test-id="expand-document-button"
+          onClick={onExpand}
+          className={actionsGroupItem}
+        ></Button>
+      )}
+      <span className={actionsGroupItemSeparator}></span>
+      {onEdit && (
+        <Button
+          size="xsmall"
+          rightGlyph={<Icon glyph="Edit"></Icon>}
+          title="Edit document"
+          data-test-id="edit-document-button"
+          onClick={onEdit}
+          className={actionsGroupItem}
+        ></Button>
+      )}
+      {onCopy && (
+        <Button
+          size="xsmall"
```

Comment:
Previously we had a little popup that showed that it was copied, should we keep this functionality with a toast or popup or is it ok like it is?
<img width="151" alt="Screen Shot 2022-03-14 at 6 06 35 PM" src="https://user-images.githubusercontent.com/1791149/158268983-5a7795f1-b6ef-4cf3-bdba-237af5e5bf1c.png">

Reply:
The tooltips are still overlapping though, right? We can use popover and add aria-live region here maybe, this will also match leafygreen copy component

Reply:
Ah yeah toasts wasn't a good suggestion from me - https://www.mongodb.design/component/toast/guidelines/ - sorry!
Popover sounds good like the leafygreen copy component 👍

Reply:
I looked at how they do this for the `Copyable` component and used a similar implementation with a tooltip https://github.com/mongodb-js/compass/pull/2897/commits/2521841f09930607e88400982dae3fe752a6aadb